quic/s2n-quic-core/src/time/timer.rs

```diff
@@ -174,6 +182,17 @@ impl Query for ArmedCount {
     }
 }
 
+impl Query for bool {
+    #[inline]
+    fn on_timer(&mut self, timer: &Timer) -> Result {
+        if timer.is_armed() {
+            *self = true;
+            return Err(QueryBreak);
+        }
+        Ok(())
+    }
+}
```

Comment:
this feels too brittle and assumes the bool is querying for `is_armed` which is not necessarily true:

```
let mut is_NOT_armed = false;
let _ = self.timers(&mut is_NOT_armed);
is_NOT_armed
```

Reply:
I can make a new type for it